metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -650,17 +650,31 @@ class TableauSourceReport(
     num_csql_field_skipped_no_name: int = 0
     num_table_field_skipped_no_name: int = 0
     # timers
-    extract_usage_stats_timer: Dict[str, float] = field(default_factory=TopKDict)
-    fetch_groups_timer: Dict[str, float] = field(default_factory=TopKDict)
-    populate_database_server_hostname_map_timer: Dict[str, float] = field(default_factory=TopKDict)
-    populate_projects_registry_timer: Dict[str, float] = field(default_factory=TopKDict)
-    emit_workbooks_timer: Dict[str, float] = field(default_factory=TopKDict)
-    emit_sheets_timer: Dict[str, float] = field(default_factory=TopKDict)
-    emit_dashboards_timer: Dict[str, float] = field(default_factory=TopKDict)
-    emit_embedded_datasources_timer: Dict[str, float] = field(default_factory=TopKDict)
-    emit_published_datasources_timer: Dict[str, float] = field(default_factory=TopKDict)
-    emit_custom_sql_datasources_timer: Dict[str, float] = field(default_factory=TopKDict)
-    emit_upstream_tables_timer: Dict[str, float] = field(default_factory=TopKDict)
+    extract_usage_stats_timer: Dict[str, float] = dataclass_field(
+        default_factory=TopKDict
+    )
+    fetch_groups_timer: Dict[str, float] = dataclass_field(default_factory=TopKDict)
+    populate_database_server_hostname_map_timer: Dict[str, float] = dataclass_field(
+        default_factory=TopKDict
+    )
+    populate_projects_registry_timer: Dict[str, float] = dataclass_field(
+        default_factory=TopKDict
+    )
+    emit_workbooks_timer: Dict[str, float] = dataclass_field(default_factory=TopKDict)
+    emit_sheets_timer: Dict[str, float] = dataclass_field(default_factory=TopKDict)
+    emit_dashboards_timer: Dict[str, float] = dataclass_field(default_factory=TopKDict)
+    emit_embedded_datasources_timer: Dict[str, float] = dataclass_field(
+        default_factory=TopKDict
+    )
+    emit_published_datasources_timer: Dict[str, float] = dataclass_field(
+        default_factory=TopKDict
+    )
+    emit_custom_sql_datasources_timer: Dict[str, float] = dataclass_field(
+        default_factory=TopKDict
+    )
+    emit_upstream_tables_timer: Dict[str, float] = dataclass_field(
```

Comment:
not a huge fan of this - does it make sense to have a `SiteReport` type, and then the `TableauSourceReport` has a `dict[str (site id), SiteReport]`?

how many sites are there reasonably going to be? the main limitation is that we don't want the report size to grow in an unbounded way

Reply:
Actually maybe this is ok - we really only care about the ones that take a long time, right